test/suites/launchTemplates/suite_test.go

```diff
@@ -0,0 +1,93 @@
+package launchtemplates
+
+import (
+	"strings"
+	"testing"
+	"regexp"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/ssm"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/aws/karpenter/pkg/apis/provisioning/v1alpha5"
+	"github.com/aws/karpenter/pkg/cloudprovider/aws/apis/v1alpha1"
+	"github.com/aws/karpenter/pkg/test"
+
+	"github.com/aws/karpenter/test/pkg/environment"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var env *environment.Environment
+
+func TestLaunchTemplates(t *testing.T) {
+	RegisterFailHandler(Fail)
+	BeforeSuite(func() {
+		var err error
+		env, err = environment.NewEnvironment(t)
+		Expect(err).ToNot(HaveOccurred())
+	})
+	RunSpecs(t, "LaunchTemplates")
+}
+
+var _ = BeforeEach(func() {
+	env.BeforeEach()
+})
+var _ = AfterEach(func() {
+	env.AfterEach()
+})
+
+// TODO: Build Custom AMI Path dynamically based on architecture, server version.
+// Avoiding the use of the recommended AMI path, since those are selected by Provisioners by default
+// and can result in false positive tests.
+var CustomAMIPath string = "/aws/service/eks/optimized-ami/1.21/amazon-linux-2/"
+
+var _ = Describe("Custom Launch Templates", func() {
+	It("should use the AMI defined by the AMI Selector", func() {
+		amiUnderTest := customAMI()
+		provider := test.AWSNodeTemplate(test.AWSNodeTemplateOptions{AWS: v1alpha1.AWS{
+			SecurityGroupSelector: map[string]string{"karpenter.sh/discovery": env.Options.ClusterName},
+			SubnetSelector:        map[string]string{"karpenter.sh/discovery": env.Options.ClusterName},
+			AMIFamily:             &v1alpha1.AMIFamilyAL2,
+		},
+			AMISelector: map[string]string{"aws-ids": amiUnderTest}, // TODO - Retrieve recommended EKS AMI and use here.
+		})
+		provisioner := test.Provisioner(test.ProvisionerOptions{ProviderRef: &v1alpha5.ProviderRef{Name: provider.Name}})
+		pod := test.Pod()
+
+		env.ExpectCreatedNodeCount("==", 0)
+		env.ExpectCreated(pod, provider, provisioner)
+
+		env.EventuallyExpectHealthy(pod)
+		env.ExpectCreatedNodeCount("==", 1)
+
+		var node v1.Node
+		env.Client.Get(env.Context, types.NamespacedName{Name:  pod.Spec.NodeName}, &node)
+		instanceID := instanceID(node.Spec.ProviderID)
+		ec2api := ec2.New(&env.AWSSession)
```

Comment:
Thoughts on baking this into the env so that it always has an EC2 client we can use?

`env.ec2api.DescribeInstances`

Reply:
Sounds good!